js/Listeners.js

```diff
@@ -183,26 +185,28 @@ Listeners.prototype.removeAll = function (event) {
  * Notifies all registered listeners that an event has occurred
  *
  * @param {String} event Event type
- * @param {Object} data The event data
- * @return {Boolean}
+ * @param {Object} payload The event data
+ * @return {Boolean} Whether to continue with the action (for events named `before*`)
  */
-Listeners.prototype.notify = function (event, data) {
+Listeners.prototype.notify = function (event, payload) {
 	var returnValue = true;
-	if (this._listeners[event] !== undefined) {
-		jQuery.each(this._listeners[event], function (i, e) {
-			if (e && typeof e === 'function') {
-				if (data && data.params && data.params === true) {
-					var r = e.apply(null, data.data);
-
-					if (r === false) {
-						returnValue = false;
-					}
+	if (this._listeners[event] === undefined) {
+		return returnValue;
+	}
 
-				} else if (e(data) === false) {
-					returnValue = false;
-				}
-			}
-		});
+	if (!payload) {
+		payload = {
+			data: null
+		};
 	}
+
+	this._listeners[event].forEach(function (listenerFn) {
+		if (typeof listenerFn !== 'function') {
+			return;
+		}
+		if (listenerFn(payload.data) === false) {
+			returnValue = false;
+		}
+	});
```

Comment:
Note that support for `data.params` has been removed. According to @mipli, this was never used, and there's no intention of starting to use it. I also think it's better to stick to a simple single argument callback.